pkg/exporters/http_exporter.go

```diff
@@ -58,186 +73,231 @@ type HTTPAlertsListSpec struct {
 	CloudMetadata apitypes.CloudMetadata  `json:"cloudMetadata"`
 }
 
-func (config *HTTPExporterConfig) Validate() error {
+// NewHTTPExporter creates a new HTTPExporter instance
+func NewHTTPExporter(config HTTPExporterConfig, clusterName, nodeName string, cloudMetadata *apitypes.CloudMetadata) (*HTTPExporter, error) {
+	if err := config.validate(); err != nil {
+		return nil, fmt.Errorf("invalid config: %w", err)
+	}
+
+	return &HTTPExporter{
+		config:      config,
+		nodeName:    nodeName,
+		clusterName: clusterName,
+		httpClient: &http.Client{
+			Timeout: time.Duration(config.TimeoutSeconds) * time.Second,
+		},
+		alertMetrics:  &alertMetrics{},
+		cloudMetadata: cloudMetadata,
+	}, nil
+}
+
+func (config *HTTPExporterConfig) validate() error {
+	if config.URL == "" {
+		return fmt.Errorf("URL is required")
+	}
+
 	if config.Method == "" {
-		config.Method = "POST"
+		config.Method = defaultMethod
 	} else if config.Method != "POST" && config.Method != "PUT" {
 		return fmt.Errorf("method must be POST or PUT")
 	}
+
 	if config.TimeoutSeconds == 0 {
-		config.TimeoutSeconds = 5
+		config.TimeoutSeconds = int(defaultTimeout.Seconds())
```

Comment:
looks weird to have a duration just to extract an int... tell Claude I would have preferred `defaultTimeoutSeconds` here

Reply:
I actually like this, it's readable.

Reply:
yeah... and it's probably too late to change the config timeoutSeconds into timeoutDuration :(